app/Components/Game/index.js

```diff
@@ -2,64 +2,71 @@ import React from 'react';
 import ChatApp from '../ChatApp';
 import Profile from '../Profile';
 import Quiz from '../Quiz';
-
+import RoleSelectionMenu from '../RoleSelectionMenu';
 import io from 'socket.io-client';
 let socket = io.connect('');
```

Comment:
This establishes the connection when this JS is initially loaded from the server.  You probably want it to establish the connection when the Game component is first mounted, so this could go in `componentDidMount` as well.